### PR TITLE
Layoutcheck: Reference image achive supports subfolder

### PR DESCRIPTION
**Pull request by martingrossmann**
# Description

To organize the reference screenshots in subfolders you can add the relative structure to the name like `my/additional/folder/WelcomePage`. 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by mbeuthan — APPROVED** [review]

File: driver-ui/src/main/java/eu/tsystems/mms/tic/testframework/layout/LayoutCheck.java [hunk 1]
```diff
@@ -481,8 +479,9 @@ private static double generateDistanceImage(
         }
 
         try {
-            // write image to given file
-            ImageIO.write(distanceImage, "PNG", new File(resultFilename));
+            // Write image to given file
+            resultFilename.toFile().getParentFile().mkdirs();
+            ImageIO.write(distanceImage, "PNG", new File(resultFilename.toAbsolutePath().toString()));
```
Comment (Zsar) [inline driver-ui/src/main/java/eu/tsystems/mms/tic/testframework/layout/LayoutCheck.java:484]: `resultFilename.toAbsolutePath().toFile()`?
https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html#toFile--

**Review by Zsar — CHANGES_REQUESTED** [review]

File: driver-ui/src/main/java/eu/tsystems/mms/tic/testframework/layout/LayoutCheck.java (hunk 1 above)
Comment (martingrossmann) [reply driver-ui/src/main/java/eu/tsystems/mms/tic/testframework/layout/LayoutCheck.java:484]: `toFile()` does not create missing folders, but I need them for `ImageIO.write()`.

Comment (Zsar) [reply driver-ui/src/main/java/eu/tsystems/mms/tic/testframework/layout/LayoutCheck.java:484]: Err... but did you not create them with the preceeding line `resultFilename.toFile().getParentFile().mkdirs()`?

Comment (martingrossmann) [reply driver-ui/src/main/java/eu/tsystems/mms/tic/testframework/layout/LayoutCheck.java:484]: Ah... I was confused... You mean
`ImageIO.write(distanceImage, "PNG", resultFilename.toAbsolutePath().toFile())`?

**Review by mbeuthan — APPROVED** [review]

**Review by Zsar — APPROVED** [review]

**Pull request merged**